src/gpu_sysman.c

```diff
@@ -824,24 +903,21 @@ static bool gpu_mems(gpu_device_t *gpu, unsigned int cache_idx) {
       continue;
     }
     /* process samples */
-    char prefix[32];
-    if (!set_mem_prefix(mems[i], prefix, sizeof(prefix))) {
+    if (!set_mem_labels(mems[i], &metric)) {
       ERROR(PLUGIN_NAME ": failed to get memory module %d properties", i);
       ok = false;
       break;
     }
-    const char *gpuname = gpu->name;
-    char vname[64];
-
+    double mem_used;
     const uint64_t mem_size = gpu->memory[0][i].size;
     if (config.samples < 2) {
       const uint64_t mem_free = gpu->memory[0][i].free;
       /* Sysman reports just memory size & free amounts => calculate used */
-      snprintf(vname, sizeof(vname), "%s-used", prefix);
-      gpu_submit(gpuname, "memory", vname, mem_size - mem_free);
-
-      snprintf(vname, sizeof(vname), "%s-free", prefix);
-      gpu_submit(gpuname, "memory", vname, mem_free);
+      mem_used = mem_size - mem_free;
+      metric.value.gauge = mem_used;
+      metric_family_metric_append(&fam_bytes, metric);
+      metric.value.gauge = mem_used / mem_size;
```

Comment:
is the mem_size guaranteed to be > 0 always?

Reply:
According to spec it seems to be:
* https://spec.oneapi.io/level-zero/latest/sysman/api.html#zes-mem-state-t
* https://spec.oneapi.io/level-zero/latest/sysman/api.html#zes-mem-properties-t

> 
> zes_mem_state_t: uint64_t size:  [out] The total allocatable memory in bytes (can be less than zes_mem_properties_t.physicalSize)
> ...
> zes_mem_properties_t: uint64_t physicalSize: [out] Physical memory size in bytes. A value of 0 indicates that this property is not known. However, a call to zesMemoryGetState() will correctly return the total size of usable memory.

But I guess there could be either kernel or sysman bugs...

Reply:
Added check for this.